agent/exec/container/controller.go

```diff
@@ -193,6 +193,13 @@ func (r *controller) Wait(pctx context.Context) error {
 				// If we get here, something has gone wrong but we want to exit
 				// and report anyways.
 				return ErrContainerDestroyed
+
+			case "health_status: unhealthy":
+				// in this case, we stop the container and report unhealthy status
+				if err := r.Shutdown(ctx); err != nil {
```

Comment:
If health check fails, I'm not sure which action we should take, `r.Shutdown` or `r.Terminate`？